backup: TestDataDriven_plpgsql_user_defined_functions failed

backup.TestDataDriven_plpgsql_user_defined_functions [failed](https://tanzanite.cluster.engflow.com/invocations/default/11802f68-0204-4eef-8f3d-9a7598f7085e?testReportRun=1&testReportShard=42&testReportAttempt=1#targets-Ly9wa2cvYmFja3VwOmJhY2t1cF90ZXN0) on master @ [b3fec61ca90095c664f6432af864f18e9946f8bb](https://github.com/cockroachdb/cockroach/commits/b3fec61ca90095c664f6432af864f18e9946f8bb):

Fatal error:

```
panic: test timed out after 40m0s
running tests:
	TestDataDriven_plpgsql_user_defined_functions (36m42s)

```
Stack: 

```
goroutine 154680720 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:617:
        query-sql [0 args]
        SELECT database_name, schema_name, routine_signature, grantee, privilege_type, is_grantable
        FROM [SHOW GRANTS ON FUNCTION add]
        ----
        db1 public add(int8, int8) admin ALL true
        db1 public add(int8, int8) root ALL true
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:624:
        exec-sql [0 args]
        SET ROLE = root
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:628:
        exec-sql [1 args]
        BACKUP INTO 'nodelocal://1/test/'
        ----
    test_server_shim.go:152: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:633:
        new-cluster [2 args]
        <no input to command>
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:637:
        exec-sql [1 args]
        RESTORE FROM LATEST IN 'nodelocal://1/test/'
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:641:
        exec-sql [1 args]
        USE db1;
        SET ROLE = u1;
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:646:
        query-sql [1 args]
        SELECT add(1, 2)
        ----
        pq: user u1 does not have EXECUTE privilege on function add
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:651:
        query-sql [1 args]
        SELECT database_name, schema_name, routine_signature, grantee, privilege_type, is_grantable
        FROM [SHOW GRANTS ON FUNCTION add]
        ----
        db1 public add(int8, int8) admin ALL true
        db1 public add(int8, int8) root ALL true
    test_server_shim.go:152: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=42</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven_plpgsql_user_defined_functions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45132

backup.TestDataDriven_plpgsql_user_defined_functions [failed](https://tanzanite.cluster.engflow.com/invocations/default/46a09723-36b8-41f3-bdc9-56d87aeaa3d2?testReportRun=1&testReportShard=42&testReportAttempt=1#targets-Ly9wa2cvYmFja3VwOmJhY2t1cF90ZXN0) on master @ [844d7637f3b4dc1275e8aa05c1cf3bbb1f59f8eb](https://github.com/cockroachdb/cockroach/commits/844d7637f3b4dc1275e8aa05c1cf3bbb1f59f8eb):

Fatal error:

```
panic: test timed out after 40m0s
running tests:
	TestDataDriven_plpgsql_user_defined_functions (36m29s)

```
Stack: 

```
goroutine 167500044 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:676:
        exec-sql [1 args]
        CREATE FUNCTION add(x INT, y INT) RETURNS INT LANGUAGE PLpgSQL AS 'BEGIN RETURN x + y; END';
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:680:
        exec-sql [1 args]
        REVOKE EXECUTE ON FUNCTION ADD FROM public;
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:684:
        exec-sql [1 args]
        SET ROLE = u1;
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:688:
        query-sql [0 args]
        SELECT add(1, 2)
        ----
        pq: user u1 does not have EXECUTE privilege on function add
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:693:
        query-sql [0 args]
        SELECT database_name, schema_name, routine_signature, grantee, privilege_type, is_grantable
        FROM [SHOW GRANTS ON FUNCTION add]
        ----
        db1 public add(int8, int8) admin ALL true
        db1 public add(int8, int8) root ALL true
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:700:
        exec-sql [0 args]
        SET ROLE = root
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:704:
        exec-sql [1 args]
        BACKUP DATABASE db1 INTO 'nodelocal://1/test/'
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:709:
        exec-sql [1 args]
        RESTORE DATABASE db1 FROM LATEST IN 'nodelocal://1/test/' WITH new_db_name = db1_new
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:713:
        exec-sql [1 args]
        USE db1_new;
        SET ROLE = u1;
        ----
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=42</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven_plpgsql_user_defined_functions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

backup.TestDataDriven_plpgsql_user_defined_functions [failed](https://tanzanite.cluster.engflow.com/invocations/default/e879b55b-74be-45c5-8361-9dc163ff2659?testReportRun=3&testReportShard=42&testReportAttempt=1#targets-Ly9wa2cvYmFja3VwOmJhY2t1cF90ZXN0) on master @ [21d1123dce5401b1ec580d4156db5db075074506](https://github.com/cockroachdb/cockroach/commits/21d1123dce5401b1ec580d4156db5db075074506):

Fatal error:

```
panic: test timed out after 40m0s
running tests:
	TestDataDriven_plpgsql_user_defined_functions (35m47s)

```
Stack: 

```
goroutine 161403776 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:664:
        exec-sql [1 args]
        CREATE DATABASE db1;
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:668:
        exec-sql [1 args]
        USE db1;
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:672:
        exec-sql [1 args]
        CREATE USER u1;
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:676:
        exec-sql [1 args]
        CREATE FUNCTION add(x INT, y INT) RETURNS INT LANGUAGE PLpgSQL AS 'BEGIN RETURN x + y; END';
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:680:
        exec-sql [1 args]
        REVOKE EXECUTE ON FUNCTION ADD FROM public;
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:684:
        exec-sql [1 args]
        SET ROLE = u1;
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:688:
        query-sql [0 args]
        SELECT add(1, 2)
        ----
        pq: user u1 does not have EXECUTE privilege on function add
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:693:
        query-sql [0 args]
        SELECT database_name, schema_name, routine_signature, grantee, privilege_type, is_grantable
        FROM [SHOW GRANTS ON FUNCTION add]
        ----
        db1 public add(int8, int8) admin ALL true
        db1 public add(int8, int8) root ALL true
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:700:
        exec-sql [0 args]
        SET ROLE = root
        ----
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=42</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven_plpgsql_user_defined_functions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

backup.TestDataDriven_plpgsql_user_defined_functions [failed](https://tanzanite.cluster.engflow.com/invocations/default/73ee2d44-b3ce-4175-bff1-4c07407643b8?testReportRun=2&testReportShard=42&testReportAttempt=1#targets-Ly9wa2cvYmFja3VwOmJhY2t1cF90ZXN0) on master @ [ccaed974f75032f1b908bca2d42145d21d0092ca](https://github.com/cockroachdb/cockroach/commits/ccaed974f75032f1b908bca2d42145d21d0092ca):

Fatal error:

```
panic: test timed out after 40m0s
running tests:
	TestDataDriven_plpgsql_user_defined_functions (36m6s)

```
Stack: 

```
goroutine 168869733 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
        exec-sql [1 args]
        BACKUP INTO 'nodelocal://1/test/'
        ----
    test_server_shim.go:152: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:633:
        new-cluster [2 args]
        <no input to command>
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:637:
        exec-sql [1 args]
        RESTORE FROM LATEST IN 'nodelocal://1/test/'
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:641:
        exec-sql [1 args]
        USE db1;
        SET ROLE = u1;
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:646:
        query-sql [1 args]
        SELECT add(1, 2)
        ----
        pq: user u1 does not have EXECUTE privilege on function add
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:651:
        query-sql [1 args]
        SELECT database_name, schema_name, routine_signature, grantee, privilege_type, is_grantable
        FROM [SHOW GRANTS ON FUNCTION add]
        ----
        db1 public add(int8, int8) admin ALL true
        db1 public add(int8, int8) root ALL true
    test_server_shim.go:152: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:661:
        new-cluster [1 args]
        <no input to command>
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:664:
        exec-sql [1 args]
        CREATE DATABASE db1;
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:668:
        exec-sql [1 args]
        USE db1;
        ----
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=42</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven_plpgsql_user_defined_functions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

backup.TestDataDriven_plpgsql_user_defined_functions [failed](https://tanzanite.cluster.engflow.com/invocations/default/f9012730-be92-45bb-a41b-a0e08397516b?testReportRun=1&testReportShard=44&testReportAttempt=1#targets-Ly9wa2cvYmFja3VwOmJhY2t1cF90ZXN0) on master @ [9354770c7c6eb5a89437068d8c6a4accf8031b67](https://github.com/cockroachdb/cockroach/commits/9354770c7c6eb5a89437068d8c6a4accf8031b67):

Fatal error:

```
panic: test timed out after 40m0s
running tests:
	TestDataDriven_plpgsql_user_defined_functions (37m26s)

```
Stack: 

```
goroutine 167711098 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:676:
        exec-sql [1 args]
        CREATE FUNCTION add(x INT, y INT) RETURNS INT LANGUAGE PLpgSQL AS 'BEGIN RETURN x + y; END';
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:680:
        exec-sql [1 args]
        REVOKE EXECUTE ON FUNCTION ADD FROM public;
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:684:
        exec-sql [1 args]
        SET ROLE = u1;
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:688:
        query-sql [0 args]
        SELECT add(1, 2)
        ----
        pq: user u1 does not have EXECUTE privilege on function add
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:693:
        query-sql [0 args]
        SELECT database_name, schema_name, routine_signature, grantee, privilege_type, is_grantable
        FROM [SHOW GRANTS ON FUNCTION add]
        ----
        db1 public add(int8, int8) admin ALL true
        db1 public add(int8, int8) root ALL true
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:700:
        exec-sql [0 args]
        SET ROLE = root
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:704:
        exec-sql [1 args]
        BACKUP DATABASE db1 INTO 'nodelocal://1/test/'
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:709:
        exec-sql [1 args]
        RESTORE DATABASE db1 FROM LATEST IN 'nodelocal://1/test/' WITH new_db_name = db1_new
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/backup/backup_test_/backup_test.runfiles/com_github_cockroachdb_cockroach/pkg/backup/testdata/backup-restore/plpgsql_user_defined_functions:713:
        exec-sql [1 args]
        USE db1_new;
        SET ROLE = u1;
        ----
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=44</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven_plpgsql_user_defined_functions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/backup/backuptestutils/BUILD.bazel
+++ b/pkg/backup/backuptestutils/BUILD.bazel
@@ -14,6 +14,7 @@ go_library(
         "//pkg/kv/kvserver",
         "//pkg/sql/sqlstats",
         "//pkg/testutils",
+        "//pkg/testutils/skip",
         "//pkg/testutils/sqlutils",
         "//pkg/testutils/testcluster",
         "//pkg/util",

--- a/pkg/backup/backuptestutils/testutils.go
+++ b/pkg/backup/backuptestutils/testutils.go
@@ -19,6 +19,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlstats"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/util"
@@ -101,6 +102,11 @@ func WithSkipInvalidDescriptorCheck() BackupTestArg {
 func StartBackupRestoreTestCluster(
 	t testing.TB, clusterSize int, args ...BackupTestArg,
 ) (*testcluster.TestCluster, *sqlutils.SQLRunner, string, func()) {
+
+	// Because the deadlock detector can increase the runtime of a test by 10-100x
+	// and has not found anything in recent memory for backup/restore tests.
+	skip.UnderDeadlock(t)
+
 	ctx := logtags.AddTag(context.Background(), "start-backup-restore-test-cluster", nil)
 	opts := backupTestOptions{}
 	for _, a := range args {

